lms/static/js/dashboard/track_events.js

```diff
@@ -8,75 +8,101 @@ var edx = edx || {};
         window.analytics.trackLink(
-            course_title_link,
-            "edx.bi.dashboard.course_title.clicked",
+            $courseTitleLink,
+            'edx.bi.dashboard.course_title.clicked',
             edx.dashboard.generateTrackProperties
         );
 
-        // Emit an event  when the "course image" is clicked.
+        // Emit an event  when the 'course image' is clicked.
         window.analytics.trackLink(
-            course_image_link,
-            "edx.bi.dashboard.course_image.clicked",
+            $courseImageLink,
+            'edx.bi.dashboard.course_image.clicked',
             edx.dashboard.generateTrackProperties
         );
 
-        // Emit an event  when the "View Course" button is clicked.
+        // Emit an event  when the 'View Course' button is clicked.
         window.analytics.trackLink(
-            enter_course_link,
-            "edx.bi.dashboard.enter_course.clicked",
+            $enterCourseLink,
+            'edx.bi.dashboard.enter_course.clicked',
             edx.dashboard.generateTrackProperties
         );
 
         // Emit an event when the options dropdown is engaged.
         window.analytics.trackLink(
-            options_dropdown,
-            "edx.bi.dashboard.course_options_dropdown.clicked",
+            $optionsDropdown,
+            'edx.bi.dashboard.course_options_dropdown.clicked',
             edx.dashboard.generateTrackProperties
         );
 
-        // Emit an event  when the "Learn about verified" link is clicked.
+        // Emit an event  when the 'Learn about verified' link is clicked.
         window.analytics.trackLink(
-            course_learn_verified,
-            "edx.bi.dashboard.verified_info_link.clicked",
+            $courseLearnVerified,
+            'edx.bi.dashboard.verified_info_link.clicked',
             edx.dashboard.generateTrackProperties
         );
 
-        // Emit an event  when the "Find Courses" button is clicked.
+        // Emit an event  when the 'Find Courses' button is clicked.
         window.analytics.trackLink(
-            find_courses_btn,
-            "edx.bi.dashboard.find_courses_button.clicked",
+            $findCoursesBtn,
+            'edx.bi.dashboard.find_courses_button.clicked',
             {
-                category: "dashboard",
+                category: 'dashboard',
                 label: null
             }
         );
 
+        // Emit an event when the 'View XSeries Details' button is clicked
+        window.analytics.trackLink(
+            $xseriesBtn,
+            'edx.bi.dashboard.xseries_cta_message.clicked',
+            edx.dashboard.generateProgramProperties
+        );
+    };
+
+    edx.dashboard.xseriesTrackMessages = function() {
+        $('.xseries-action .btn').each(function(i, element) {
+            var data = edx.dashboard.generateProgramProperties($(element));
+
+            window.analytics.track( 'edx.bi.dashboard.xseries_cta_message.viewed', data );
```

Comment:
@rlucioni this was discussed at the start of the work and it was deemed sufficient to emit the events on page load instead of a more complex JS scroll listener emitting events when the elements are visible in the view port.